NVDA now has the ability to report indents as tones, speech, or both. #5906

derekriemer opened the pull request:
This is controlled by a combo box in document formatting.
Fixes #5906.

Commit:
NVDA now has the ability to report indents as tones, speech, or both.
This is controlled by a combo box in document formatting.

jcsteh commented on source/config/__init__.py:
```diff
@@ -19,6 +19,10 @@
 
+#Constants for indent tones and speech.
+INDENT_SPEECH = 1
+INDENT_TONES = 2
```
nit: I totally get why you'd put these constants here, but I think I'd prefer to see them in the speech module. While we have to define the config spec here, I'd prefer to keep subsystem specific stuff with the subsystem as much as possible.

jcsteh commented on source/gui/settingsDialogs.py:
```diff
@@ -1116,7 +1116,28 @@ def makeSettings(self, settingsSizer):
 		#
 		self.lineIndentationCheckBox=wx.CheckBox(self,wx.NewId(),label=_("Report l&ine indentation"))
 		self.lineIndentationCheckBox.SetValue(config.conf["documentFormatting"]["reportLineIndentation"])
+		self.lineIndentationCheckBox.Bind(wx.EVT_CHECKBOX,self.onReportIndent)
```
nit: Maybe `self.onReportIndentChange` to be clearer? We don't do `onBlahClick` for buttons, but that's because an "action" seems natural without the clarification.

jcsteh commented on source/gui/settingsDialogs.py:
```diff
@@ -1116,7 +1116,28 @@ def makeSettings(self, settingsSizer):
 		self.lineIndentationCheckBox.SetValue(config.conf["documentFormatting"]["reportLineIndentation"])
+		self.lineIndentationCheckBox.Bind(wx.EVT_CHECKBOX,self.onReportIndent)
 		settingsSizer.Add(self.lineIndentationCheckBox,border=10,flag=wx.BOTTOM)
+		self.indentTypeSizer = sizer=wx.BoxSizer(wx.HORIZONTAL)
```
nit: No need to save this on self. You don't reference it later.

jcsteh commented on source/gui/settingsDialogs.py:
```diff
@@ -1176,6 +1197,12 @@ def makeSettings(self, settingsSizer):
+		if self.lineIndentationCheckBox.GetValue():
+			self.indentTypeList.Enable(True)
+		else:
+			self.indentTypeList.Enable(False)
```
nit: IMO, this is shorter/more readable:

```
self.indentTypeList.Enabled = self.lineIndentationCheckBox.IsChecked()
```

derekriemer replied:
Um, that overrides the method's name. Is this possible I.E. = rather 
than ( )


On 6/9/2016 4:40 AM, James Teh wrote:
>
> In source/gui/settingsDialogs.py 
> <https://github.com/nvaccess/nvda/pull/6057#discussion_r66418495>:
>
> > @@ -1176,6 +1197,12 @@ def makeSettings(self, settingsSizer):
> >  	def postInit(self):
> >  		self.detectFormatAfterCursorCheckBox.SetFocus()
> >
> > +	def onReportIndent(self, evt):
> > +		if self.lineIndentationCheckBox.GetValue():
> > +			self.indentTypeList.Enable(True)
> > +		else:
> > +			self.indentTypeList.Enable(False)
>
> nit: IMO, this is shorter/more readable:
>
> |self.indentTypeList.Enabled = self.lineIndentationCheckBox.IsChecked() |
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/6057/files/149618ac951d0380dad97bf4bd02ae1f00e451f9#r66418495>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivYJIFtF6gRUv8b5RzRJJzpq8RcNpks5qJ-2sgaJpZM4IxqMD>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

jcsteh replied:
.Enabled is a property, so you can get/set it. Under the hood, it calls Enable(). Either way is fine; I just find properties nicer to read personally, but no need to change this now in your new code.

jcsteh commented on source/speech.py:
```diff
@@ -408,23 +408,28 @@ def splitTextIndentation(text):
 RE_INDENTATION_CONVERT = re.compile(r"(?P<char>\s)(?P=char)*", re.UNICODE)
+BASE_FREQUENCY = 220 #One octave below middle A.
+TONE_DURATION = 80 #Milleseconds
+MAX_SPACES = 72
```
nit: Please prefix these so it's clear they're related to indentation when looking at the namespace. Perhaps an `INDTONE_` prefix; e.g. `INDTONE_MAX_SPACES`?

jcsteh commented on source/speech.py:
```diff
@@ -436,8 +441,16 @@ def getIndentationSpeech(indentation):
 			res.append(u"{count} {symbol}".format(count=count, symbol=symbol))
-
-	return " ".join(res)
+		quarterTones += (count*4 if raw[0]== "\t" else count)
```
Out of interest, why the `[0]`? At this point, we should already be dealing with single characters... and even if we weren't in some future/alternative universe, that suggests the space is a double byte character or something, so the second character could be important.

derekriemer replied:
This code uses raw[0] and len(raw) which is suggesting raw is a list.

raw = m.group()
         symbol = characterProcessing.processSpeechSymbol(locale, raw[0])
         count = len(raw)
         if symbol == raw[0]:


On 6/9/2016 4:44 AM, James Teh wrote:
>
> In source/speech.py 
> <https://github.com/nvaccess/nvda/pull/6057#discussion_r66418995>:
>
> > @@ -436,8 +441,16 @@ def getIndentationSpeech(indentation):
> >  			res.append(symbol)
> >  		else:
> >  			res.append(u"{count} {symbol}".format(count=count, symbol=symbol))
> > -
> > -	return " ".join(res)
> > +		quarterTones += (count*4 if raw[0]== "\t" else count)
>
> Out of interest, why the |[0]|? At this point, we should already be 
> dealing with single characters... and even if we weren't in some 
> future/alternative universe, that suggests the space is a double byte 
> character or something, so the second character could be important.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/6057/files/149618ac951d0380dad97bf4bd02ae1f00e451f9#r66418995>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivZoxeDuVjabSpTjZYxA_GKouq6R5ks5qJ-6igaJpZM4IxqMD>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

jcsteh replied:
My apologies. You're quite correct. That's what I get for code reviewing late at night. :)

jcsteh commented on source/speech.py:
```diff
@@ -436,8 +441,16 @@ def getIndentationSpeech(indentation):
+		quarterTones += (count*4 if raw[0]== "\t" else count)
+
+		speak = indentConfig & config.INDENT_SPEECH
+		if indentConfig & config.INDENT_TONES: #Remove me during speech refactor.
```
We won't be removing this bit during speech refactor. The only thing that will happen is tones.beep will be changed to return a `SpeechCommand`. So, perhaps move the comment down to the tones.beep.

jcsteh commented on source/speech.py:
```diff
@@ -436,8 +441,16 @@ def getIndentationSpeech(indentation):
+			if quarterTones <= MAX_SPACES:
+				pitch = BASE_FREQUENCY*2**(quarterTones/24.0)
+				tones.beep(pitch, TONE_DURATION)
+			else: #we have more than 72 spaces (24 tabs), and must speak it since we don't want to hurt the users ears.
```
nit: Great comment, but please move it inside the block. We sometimes put tiny comments on the else line, but I think a comment like this makes more sense inside the block.

jcsteh commented on source/speech.py:
```diff
@@ -436,8 +441,16 @@ def getIndentationSpeech(indentation):
 			res.append(symbol)
```
It just occurred to me that if we're not going to speak, there's no need to call characterProcessing, etc. We could optimise that out. See if you can do this, but don't bother if the code becomes too unreadable as a result (e.g. too many if checks).

derekriemer replied:
No. We have to process now, because we might have more than 72 spaces 
(18 tabs). I might be able to figure out a way but it'd be messy.


On 6/9/2016 4:50 AM, James Teh wrote:
>
> In source/speech.py 
> <https://github.com/nvaccess/nvda/pull/6057#discussion_r66419846>:
>
> > @@ -436,8 +441,16 @@ def getIndentationSpeech(indentation):
> >  			res.append(symbol)
>
> It just occurred to me that if we're not going to speak, there's no 
> need to call characterProcessing, etc. We could optimise that out. See 
> if you can do this, but don't bother if the code becomes too 
> unreadable as a result (e.g. too many if checks).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/6057/files/149618ac951d0380dad97bf4bd02ae1f00e451f9#r66419846>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivZ7qBSAmpOk1B5538PjiYo02RO0Cks5qJ-_dgaJpZM4IxqMD>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

jcsteh commented on user_docs/en/userGuide.t2t:
```diff
@@ -1213,13 +1213,28 @@ You can configure reporting of:
 To toggle these settings from anywhere, please assign custom gestures using the [Input Gestures dialog #InputGestures].
 
+
+
```
nit: Added extraneous blank lines here. Please remove.

Commit:
fix code review comments from @jcsteh

jcsteh commented on user_docs/en/userGuide.t2t:
```diff
@@ -1220,6 +1220,19 @@ By default, NVDA will detect the formatting at the position of the System caret
+- Tones
+- Speech
+- Both Speech and Tones
+
```
nit: Change this list ending to a dash on a line by itself rather than a blank line:

`-`

In changes.t2t, we use a double blank line because it's more natural when there are only lists, but in the User Guide, we use the dash method because lists are interspersed with other text, so it's clearer.

jcsteh commented on user_docs/en/userGuide.t2t:
```diff
@@ -1220,6 +1220,19 @@ By default, NVDA will detect the formatting at the position of the System caret
+- Both Speech and Tones
+
+
+If speech is selected, when the level of indented text changes, NVDA will say something like "twelve spaces" or "four tabs."
```
nit: That should be: "12 space" and "4 tab". Note the numerals and the singular nouns. Obviously, plural would be better, but having plural definitions for all symbols is overkill, so we don't.

jcsteh commented on user_docs/en/userGuide.t2t:
```diff
@@ -1220,6 +1220,19 @@ By default, NVDA will detect the formatting at the position of the System caret
+If speech is selected, when the level of indented text changes, NVDA will say something like "twelve spaces" or "four tabs."
+
+If Tones is selected, when the level of indented text changes, tones indicate the amount of change in indent.
+The tone will increase in pitch every space, and for a tab, it will increase in pitch the equivalent of 4 spaces.
```
I'm wondering whether these descriptions of speech and tones should be moved up into the list. It seems redundant to have the list and then the descriptions in prose below it.

Commit:
fix comments made about user guide.

derekriemer commented:
Done.

jcsteh commented:
#5906 is now incubated. Thanks for your work... and good job. :)

One final request for the future: please reference the issue in the comment when creating a PR. Note that if you reference it in the commit, it will be included in the default PR comment.

Commit:
NVDA now properly speaks "no indent" only while speech is turned on. re #5906

Commit:
This is now controlled by two text boxes.
This no longer beeps with speech mode off. re #5906 @jcsteh I haven't had time to update the user guide yet, I'll let you know when that's complete.

Commit:
Make the gui One list. Re: #5906 CC @jcsteh

Commit:
User guide reflects code changes. re: #5906
@jcsteh considder this ready for further scrupulation.

jcsteh commented on source/config/__init__.py:
```diff
@@ -178,6 +178,7 @@ def validateConfig(configObj,validator,validationResult=None,keyList=None):
 	reportPage = boolean(default=true)
 	reportLineNumber = boolean(default=False)
 	reportLineIndentation = boolean(default=False)
+	reportToneIndentation = boolean(default=False)
```
I think we should rename this to `reportLineIndentationWithTones` or `lineIndentationTones`, since it isn't clear that this is related to line indentation (as opposed to paragraph indentation).

jcsteh commented on source/gui/settingsDialogs.py:
```diff
@@ -1111,12 +1111,27 @@ def makeSettings(self, settingsSizer):
-		settingsSizer.Add(self.lineIndentationCheckBox,border=10,flag=wx.BOTTOM)
+		sizer=wx.BoxSizer(wx.HORIZONTAL)
+		# Translators: This is the label for a combobox in the
+		# Document  Formatting  dialog (possible choices are Off, Speech, Tones, or Both.
```
nit: The translator comment specifies the choices, but doesn't specify what they're for. I realise they can figure this out from the text. Still, I think it'd be clearer to say something like:

> This is the label for a combobox in the Document Formatting dialog wherein the user can choose how line indentation is reported (off, speech, tones or both).

jcsteh commented on source/gui/settingsDialogs.py:
```diff
@@ -1111,12 +1111,27 @@ def makeSettings(self, settingsSizer):
+		sizer=wx.BoxSizer(wx.HORIZONTAL)
+		# Translators: This is the label for a combobox in the
+		# Document  Formatting  dialog (possible choices are Off, Speech, Tones, or Both.
+		sizer.Add(wx.StaticText(self,wx.ID_ANY,label=_("Report Line &Indents With:")))
```
nit: Only the first word "Report" should be capitalised.

jcsteh replied:
Hmm, and this should probably be "indentation", not "indents".

jcsteh commented on source/gui/settingsDialogs.py:
```diff
@@ -1111,12 +1111,27 @@ def makeSettings(self, settingsSizer):
+			_("Tones"),
+			#Translators: A setting to report indents with both tones and Speech.
+			_("Both  Speech and Tones")
+		]
```
@josephsl: Do you think we're going to need a gettext context here? Are there conflicting messages where this could cause ambiguity?

derekriemer replied:
@josephsl:  just making sure you saw this. :-)


On 7/1/2016 7:00 AM, James Teh wrote:
>
> In source/gui/settingsDialogs.py 
> <https://github.com/nvaccess/nvda/pull/6057#discussion_r69295674>:
>
> > -		self.lineIndentationCheckBox.SetValue(config.conf["documentFormatting"]["reportLineIndentation"])
> > -		settingsSizer.Add(self.lineIndentationCheckBox,border=10,flag=wx.BOTTOM)
> > +		sizer=wx.BoxSizer(wx.HORIZONTAL)
> > +		# Translators: This is the label for a combobox in the
> > +		# Document  Formatting  dialog (possible choices are Off, Speech, Tones, or Both.
> > +		sizer.Add(wx.StaticText(self,wx.ID_ANY,label=_("Report Line &Indents With:")))
> > +		indentChoices=[
> > +			#Translators: A setting to report No  line Indentation.
> > +			_("Off"),
> > +			#Translators: A setting to report indents with Speech.
> > +			_("Speech"),
> > +			#Translators: A setting to report indents with tones.
> > +			_("Tones"),
> > +			#Translators: A setting to report indents with both tones and Speech.
> > +			_("Both  Speech and Tones")
> > +		]
>
> @josephsl <https://github.com/josephsl>: Do you think we're going to 
> need a gettext context here? Are there conflicting messages where this 
> could cause ambiguity?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/6057/files/6d6ee04d8149a704f5d806a0712cbad3fdd04cdc#r69295674>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivQ27avFRKx6Coo8sBVZRzMG4AGNPks5qRQ9wgaJpZM4IxqMD>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

jcsteh commented on source/gui/settingsDialogs.py:
```diff
@@ -1111,12 +1111,27 @@ def makeSettings(self, settingsSizer):
+			_("Both  Speech and Tones")
+		]
+		self.lineIndentationCombo = wx.Choice(self,wx.ID_ANY,choices=indentChoices)
+		#This is for Speech. For legacy reasons, we leave this name.
```
I'm guessing this comment refers to the reportLineIndentation key, but given that the code below refers to both keys, this is unclear. Please clarify or remove.

jcsteh commented on source/gui/settingsDialogs.py:
```diff
@@ -1190,7 +1205,19 @@ def onOk(self,evt):
+			config.conf["documentFormatting"]["reportToneIndentation"] = True
+		if choice == 3:
+			config.conf["documentFormatting"]["reportLineIndentation"] = True
+			config.conf["documentFormatting"]["reportToneIndentation"] = True
```
You can simplify this down to two lines:

```
config.conf["documentFormatting"]["reportLineIndentation"] = choice in (1, 3)
config.conf["documentFormatting"]["reportToneIndentation"] = choice in (2, 3)
```

Brian1Gaff replied:
Blimey, my Boolean brain just exploded....
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message ----- 
From: "James Teh" <notifications@github.com>
To: "nvaccess/nvda" <nvda@noreply.github.com>
Sent: Friday, July 01, 2016 11:21 PM
Subject: Re: [nvaccess/nvda] NVDA now has the ability to report indents as 
tones, speech, or both. (#6057)


>> @@ -1190,7 +1205,19 @@ def onOk(self,evt):
>> 
>> config.conf["documentFormatting"]["reportGrammarErrors"]=self.grammarErrorsCheckBox.IsChecked()
>> 
>> config.conf["documentFormatting"]["reportPage"]=self.pageCheckBox.IsChecked()
>> 
>> config.conf["documentFormatting"]["reportLineNumber"]=self.lineNumberCheckBox.IsChecked()
>> - 
>> config.conf["documentFormatting"]["reportLineIndentation"]=self.lineIndentationCheckBox.IsChecked()
>> + choice = self.lineIndentationCombo.GetSelection()
>> + if choice == 0:
>> + config.conf["documentFormatting"]["reportLineIndentation"] = False
>> + config.conf["documentFormatting"]["reportToneIndentation"] = False
>> + elif choice == 1:
>> + config.conf["documentFormatting"]["reportLineIndentation"] = True
>> + config.conf["documentFormatting"]["reportToneIndentation"] = False
>> + elif choice == 2:
>> + config.conf["documentFormatting"]["reportLineIndentation"] = False
>> + config.conf["documentFormatting"]["reportToneIndentation"] = True
>> + if choice == 3:
>> + config.conf["documentFormatting"]["reportLineIndentation"] = True
>> + config.conf["documentFormatting"]["reportToneIndentation"] = True
>
> You can simplify this down to two lines:
>
> ```
> config.conf["documentFormatting"]["reportLineIndentation"] = choice in (1, 
> 3)
> config.conf["documentFormatting"]["reportToneIndentation"] = choice in (2, 
> 3)
> ```
>
> ---
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/pull/6057/files/6d6ee04d8149a704f5d806a0712cbad3fdd04cdc#r69362189

derekriemer replied:
HMMMMMMMMMMMMMM?


On 7/2/2016 3:36 AM, Brian1Gaff wrote:
>
> In source/gui/settingsDialogs.py 
> <https://github.com/nvaccess/nvda/pull/6057#discussion_r69375799>:
>
> > @@ -1190,7 +1205,19 @@ def onOk(self,evt):
> >  		config.conf["documentFormatting"]["reportGrammarErrors"]=self.grammarErrorsCheckBox.IsChecked()
> >  		config.conf["documentFormatting"]["reportPage"]=self.pageCheckBox.IsChecked()
> >  		config.conf["documentFormatting"]["reportLineNumber"]=self.lineNumberCheckBox.IsChecked()
> > -		config.conf["documentFormatting"]["reportLineIndentation"]=self.lineIndentationCheckBox.IsChecked()
> > +		choice = self.lineIndentationCombo.GetSelection()
> > +		if choice == 0:
> > +			config.conf["documentFormatting"]["reportLineIndentation"] = False
> > +			config.conf["documentFormatting"]["reportToneIndentation"] = False
> > +		elif choice == 1:
> > +			config.conf["documentFormatting"]["reportLineIndentation"] = True
> > +			config.conf["documentFormatting"]["reportToneIndentation"] = False
> > +		elif choice == 2:
> > +			config.conf["documentFormatting"]["reportLineIndentation"] = False
> > +			config.conf["documentFormatting"]["reportToneIndentation"] = True
> > +		if choice == 3:
> > +			config.conf["documentFormatting"]["reportLineIndentation"] = True
> > +			config.conf["documentFormatting"]["reportToneIndentation"] = True
> Blimey, my Boolean brain just exploded.... Brian 
> bglists@blueyonder.co.uk Sent via blueyonder. Please address personal 
> email to:- briang1@blueyonder.co.uk, putting 'Brian Gaff' in the 
> display name field. ----- Original Message ----- From: "James Teh" 
> <notifications@github.com> To: "nvaccess/nvda" 
> <nvda@noreply.github.com> Sent: Friday, July 01, 2016 11:21 PM 
> Subject: Re: [nvaccess/nvda] NVDA now has the ability to report 
> indents as tones, speech, or both. (#6057 
> <https://github.com/nvaccess/nvda/pull/6057>)
> … <#>
> > @@ -1190,7 +1205,19 @@ def onOk(self,evt): > > 
> config.conf["documentFormatting"]["reportGrammarErrors"]=self.grammarErrorsCheckBox.IsChecked()> 
> > 
> config.conf["documentFormatting"]["reportPage"]=self.pageCheckBox.IsChecked()> 
> > 
> config.conf["documentFormatting"]["reportLineNumber"]=self.lineNumberCheckBox.IsChecked()> 
> - > 
> config.conf["documentFormatting"]["reportLineIndentation"]=self.lineIndentationCheckBox.IsChecked()> 
> + choice = self.lineIndentationCombo.GetSelection() > + if choice == 
> 0: > + config.conf["documentFormatting"]["reportLineIndentation"] = 
> False > + config.conf["documentFormatting"]["reportToneIndentation"] = 
> False > + elif choice == 1: > + 
> config.conf["documentFormatting"]["reportLineIndentation"] = True > + 
> config.conf["documentFormatting"]["reportToneIndentation"] = False > + 
> elif choice == 2: > + 
> config.conf["documentFormatting"]["reportLineIndentation"] = False > + 
> config.conf["documentFormatting"]["reportToneIndentation"] = True > + 
> if choice == 3: > + 
> config.conf["documentFormatting"]["reportLineIndentation"] = True > + 
> config.conf["documentFormatting"]["reportToneIndentation"] = True You 
> can simplify this down to two lines: ``` 
> config.conf["documentFormatting"]["reportLineIndentation"] = choice in 
> (1, 3) config.conf["documentFormatting"]["reportToneIndentation"] = 
> choice in (2, 3) ``` --- You are receiving this because you are 
> subscribed to this thread. Reply to this email directly or view it on 
> GitHub: 
> https://github.com/nvaccess/nvda/pull/6057/files/6d6ee04d8149a704f5d806a0712cbad3fdd04cdc#r69362189 
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/6057/files/6d6ee04d8149a704f5d806a0712cbad3fdd04cdc#r69375799>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivaZZCGX0BgvgOEakh0fLY9CYXPhrks5qRjELgaJpZM4IxqMD>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

jcsteh commented on user_docs/en/userGuide.t2t:
```diff
@@ -1220,6 +1220,16 @@ By default, NVDA will detect the formatting at the position of the System caret
 
 Enable this option while proof reading documents in applications such as Microsoft Word, where formatting is important.
 
+==== Report line indents With ====
```
nit: Capital w in "with" should be lower case.

jcsteh commented on user_docs/en/userGuide.t2t:
```diff
@@ -1220,6 +1220,16 @@ By default, NVDA will detect the formatting at the position of the System caret
 Enable this option while proof reading documents in applications such as Microsoft Word, where formatting is important.
 
+==== Report line indents With ====
+the Report line indents with combo box has four options.
```
I feel this introductory sentence should possibly explain the point of the choice here; e.g. "This option allows you to configure how indentation at the beginning of liens is reported."

jcsteh commented on user_docs/en/userGuide.t2t:
```diff
@@ -1220,6 +1220,16 @@ By default, NVDA will detect the formatting at the position of the System caret
+==== Report line indents With ====
+the Report line indents with combo box has four options.
+
+- Off: NVDA will not treat indents in text specially.
```
Again, be careful with the distinction between line indentation and paragraph indentation here. Because this is within the line indentation section, you can probably just change "indents in text" to "indentation".

jcsteh commented:
Okay. I'm done now. :) Thanks for your work.

derekriemer commented:
I really botched this. Hold up

Commit:
Fix Code Review. cc @jcsteh

derekriemer commented:
Fixed

jcsteh commented on source/gui/settingsDialogs.py:
```diff
@@ -1112,23 +1112,22 @@ def makeSettings(self, settingsSizer):
 		settingsSizer.Add(self.lineNumberCheckBox,border=10,flag=wx.BOTTOM)
 		sizer=wx.BoxSizer(wx.HORIZONTAL)
-		# Translators: This is the label for a combobox in the
+		# Translators: This is the label for a combobox controlling the line indentation in the
```
nit: "controlling the line indentation" -> "controlling the reporting of line indentation"

jcsteh commented on source/gui/settingsDialogs.py:
```diff
@@ -1112,23 +1112,22 @@ def makeSettings(self, settingsSizer):
+		# Translators: This is the label for a combobox controlling the line indentation in the
 		# Document  Formatting  dialog (possible choices are Off, Speech, Tones, or Both.
-		sizer.Add(wx.StaticText(self,wx.ID_ANY,label=_("Report Line &Indents With:")))
+		sizer.Add(wx.StaticText(self,wx.ID_ANY,label=_("Report Line &Indentation With:")))
```
nit: Lower case for "line indentation". I think you missed this in the last review. :)

jcsteh commented on user_docs/en/userGuide.t2t:
```diff
@@ -1220,12 +1220,12 @@ By default, NVDA will detect the formatting at the position of the System caret
 
-==== Report line indents With ====
-the Report line indents with combo box has four options.
+==== Report line indents with ====
```
indents -> indentation

jcsteh commented on user_docs/en/userGuide.t2t:
```diff
@@ -1220,12 +1220,12 @@ By default, NVDA will detect the formatting at the position of the System caret
-==== Report line indents With ====
-the Report line indents with combo box has four options.
+==== Report line indents with ====
+This option allows you to configure how indentation at the beginning of lines is reported. The Report line indents with combo box has four options.
```
nit: Please split these two sentences onto separate lines. We do one sentence per line in our documentation because it makes life easier for translators.

Commit:
More review complete cc @jcsteh

jcsteh commented on user_docs/en/userGuide.t2t:
```diff
@@ -1220,8 +1220,9 @@ By default, NVDA will detect the formatting at the position of the System caret
-This option allows you to configure how indentation at the beginning of lines is reported. The Report line indents with combo box has four options.
+==== Report line indentation with ====
+This option allows you to configure how indentation at the beginning of lines is reported.
+The Report line indents with combo box has four options.
```
You missed another instance of "Report line indents with" -> "Report line indentation with". :)

Commit:
Fixed all "indents" -> indentation

jcsteh commented:
What's New entry: in New Features:

```
- NVDA can now indicate line indentation using tones. This can be configured using the "Report line indentation with" combo box in NVDA's Document Formatting preferences dialog. (#5906)
```

josephsl commented:
Hi,

Found a major inconsistency/bug: if report indentation is off, pressing NVDA+F from an indented text does not announce indentation level.

STR:

1. Open Notepad++ or similar editor.
2. Using latest next build, set report indentation to off.
3. Type something into the editor and indent it.
4. Press NVDA+F.

Expected: NVDA announces indent level and type.
Actual: No announcement of indents.
Workaround: Need to set report indentation to speech, tone or both.

Note that this affects NVDA+F (report formatting). Thanks.

derekriemer commented:
Wow, that happens because this didn't use to touch the config. Anyway, 
I'll fix this.


On 7/6/2016 6:17 PM, Joseph Lee wrote:
>
> Hi,
>
> Found a major inconsistency/bug: if report indentation is off, 
> pressing NVDA+F from an indented text does not announce indentation level.
>
> STR:
>
>  1. Open Notepad++ or similar editor.
>  2. Using latest next build, set report indentation to off.
>  3. Type something into the editor and indent it.
>  4. Press NVDA+F.
>
> Expected: NVDA announces indent level and type.
> Actual: No announcement of indents.
> Workaround: Need to set report indentation to speech, tone or both.
>
> Note that this affects NVDA+F (report formatting). Thanks.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/6057#issuecomment-230945295>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivX3HeFS7nUlUTrHFj9N5cEL7wwljks5qTEWXgaJpZM4IxqMD>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

Commit:
Fix bug with report formatting script.

Commit:
Fix copyright header.

Commit:
Fix indentation error.
* The code for playing a tone was actually surprisingly indented two tabs, rahter than one tab. It worked in all cases except a few, I.E. 18 tabs, and then 3 spaces.
This is now fixed, and indents work as expected.

Commit:
remove grammar errors.

Commit:
Fix merge conflicts.

Commit:
Merge remote-tracking branch 'nvaccess/master' into i5906

derekriemer commented:
@MichaelDCurran Is this going to incubate into next in time for 16.3?


On 7/6/2016 6:17 PM, Joseph Lee wrote:
>
> Hi,
>
> Found a major inconsistency/bug: if report indentation is off, 
> pressing NVDA+F from an indented text does not announce indentation level.
>
> STR:
>
>  1. Open Notepad++ or similar editor.
>  2. Using latest next build, set report indentation to off.
>  3. Type something into the editor and indent it.
>  4. Press NVDA+F.
>
> Expected: NVDA announces indent level and type.
> Actual: No announcement of indents.
> Workaround: Need to set report indentation to speech, tone or both.
>
> Note that this affects NVDA+F (report formatting). Thanks.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/6057#issuecomment-230945295>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivX3HeFS7nUlUTrHFj9N5cEL7wwljks5qTEWXgaJpZM4IxqMD>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

michaelDCurran commented:
Sorry: but now NVDA+f does not work. reportLineSpacing is missing from the reportFormatting dict.

Commit:
Fix broken lineSpacing in formatConfig

derekriemer commented:
Fixed, I'm sorry I missed that.


On 7/17/2016 6:51 PM, Michael Curran wrote:
>
> Sorry: but now NVDA+f does not work. reportLineSpacing is missing from 
> the reportFormatting dict.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/6057#issuecomment-233214087>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFGivWGUvts7Bjj7tNcwjAEWbjpnaONkks5qWs4QgaJpZM4IxqMD>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194
